Ruby on Rails Capstone

VitorGuedesMadeira opened the pull request:
### Ruby on Rails capstone project - Budget app
[Project Requirements](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md)

[Video description](https://www.loom.com/share/4ef30a8d4c81435b97bc2c076be844db)

**_Learning Objectives:_**
✔️ Use ruby gems as software packages system.
✔️ Install Ruby on Rails framework.
✔️ Understand Rails RESTful design and router.
✔️ Use controllers to handle requests and render empty views.
✔️ Use params from browser request in a safe way.
✔️ Use preprocessed html file with embedded Ruby code.
✔️ Use layouts and templates for shared content.
✔️ Use database migration files to maintain database schema.
✔️ Use validations for models.
✔️ Secure app from n+1 problems.
✔️ Understand what ORM is.
✔️ Write SQL queries with ActiveRecord.
✔️ Set up associations between models.
✔️ Build a webapp that requires the user to log in.
✔️ Use devise gem for authentication.
✔️ Limit access to webapp resources based on authorization rules.
✔️ Analyze in writing why you have made a coding choice using one structure over another.

**_Author:_** [Vitor Guedes Madeira](https://github.com/VitorGuedesMadeira) 👦 💻 
<img width="374" alt="erd_diagram" src="https://user-images.githubusercontent.com/103270117/202256043-5e3a9294-59c2-45e4-b1fb-52fd293c81a1.png">
<img width="309" alt="transactions_list" src="https://user-images.githubusercontent.com/103270117/202256077-ebc61584-9ce6-4a12-958e-8ada4ab98466.png">

VitorGuedesMadeira committed:
Add database

VitorGuedesMadeira committed:
Add rubocop and stylinters

VitorGuedesMadeira committed:
Add user scaffold

VitorGuedesMadeira committed:
Add establishment scaffold

VitorGuedesMadeira committed:
Add foreign keys

VitorGuedesMadeira committed:
Add splash screen controller and define routes

VitorGuedesMadeira committed:
Reset database

VitorGuedesMadeira committed:
Add autentication

VitorGuedesMadeira committed:
Add associations

VitorGuedesMadeira committed:
Add validations to models

VitorGuedesMadeira committed:
Add splash redirection route

VitorGuedesMadeira committed:
Add index/show views

VitorGuedesMadeira committed:
Add changes

VitorGuedesMadeira committed:
Add changes to views

VitorGuedesMadeira committed:
Add cancancan

VitorGuedesMadeira committed:
Add authorization for establishments

VitorGuedesMadeira committed:
Add awesome fonts

VitorGuedesMadeira committed:
Add buttons styles

VitorGuedesMadeira committed:
Add rspec-rails

VitorGuedesMadeira committed:
Add unit testings

VitorGuedesMadeira committed:
Add styles

VitorGuedesMadeira committed:
Add integration test for sign in

VitorGuedesMadeira committed:
Add styles

VitorGuedesMadeira committed:
Add request tests and styles

VitorGuedesMadeira committed:
Fix linters

VitorGuedesMadeira committed:
Fix css linters

VitorGuedesMadeira committed:
Add alerts

VitorGuedesMadeira committed:
Merge pull request #1 from VitorGuedesMadeira/testings

Testings

VitorGuedesMadeira committed:
Add changes

VitorGuedesMadeira committed:
Fix linux error

VitorGuedesMadeira committed:
Change readme file

VitorGuedesMadeira committed:
Fix readme file

VitorGuedesMadeira committed:
Add video description

VitorGuedesMadeira committed:
Change readme file

sja-thedude reviewed (changes requested):
## Required Changes ♻️

Hi @VitorGuedesMadeira, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![768070f3-bc9a-4aba-b8bc-25793e00ea1e_text](https://user-images.githubusercontent.com/70269507/159483175-2f61461d-b352-4cb4-98f8-b173e05ab5ff.gif)

## Highlights
✔️ Professional Readme
✔️ Correct Gitflow
✔️ All linters passing

## Required 
- [ ] Kindly make sure the loom video is recorded with your video on, also the video contains a highlighted piece of the code the student finds interesting or is proud of making.
- [ ] Kindly make sure to fix the error which occurred while running tests.
<img width="636" alt="Screenshot 2022-11-16 at 10 03 45 PM" src="https://user-images.githubusercontent.com/70269507/202273176-45e7034d-9065-4a3e-a4dd-1d03cda34571.png">

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VitorGuedesMadeira committed:
Fix rspec

VitorGuedesMadeira commented:
@sja-thedude Thanks for the feedback! I already fixed the test problem. I removed an alert after deploying it so it resulted in the testing error since I was expecting to see it. About the video, I actually tried to use it but I was facing some camera issues here. So I just applied to it since the camera on was not a must on the project requirements 😄

mukhammadrizooff reviewed (changes requested):
Hi @VitorGuedesMadeira, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

<img src="https://media4.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 
- Descriptive `PR` description and commit history ✔️ 

## Suggested changes
- [ ] [REQUIRED] Good job so far👍 You did a great job, but I think you forgot to fulfill one of the requirements that the [previous reviewer](https://github.com/VitorGuedesMadeira/Budget-App/pull/2#pullrequestreview-1183198838) mentioned. Accordingly, in your [presentation video](https://www.loom.com/share/4ef30a8d4c81435b97bc2c076be844db), `highlight about a piece of the code in the project you find interesting or proud of making in your codebase`. You have made a great presentation video but the above mentioned `requirement` is not fulfilled in the video, please `re-record` your project presentation video with the given requirement and update the link on your `PR` description when you will have finish 🙏🙏🙏 

![image](https://user-images.githubusercontent.com/63915024/202284826-c1350bb3-8249-45fb-9a14-343cb67e8382.png)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

VitorGuedesMadeira committed:
Add uniqueness

bimbolabuari reviewed (commented):
Hello @VitorGuedesMadeira 👋,

_This is an additional review as a result of the second opinion request._

- Kudos for working on your capstone project 👍
   -  The CRs were right, we explicitly stated in the code review guidelines and project requirement that there should be a highlight(emphasize) about a piece of the code the student finds interesting or is proud of making. While you implemented a unique feature of user update and mention it in the video, it would have been nice to emphasize it as an implementation you are proud of, this is just to check for our attention to detail 🙂 . 
   
  - @VitorGuedesMadeira, you have done an awesome job 🥇,  and have a unique implementation you stated in the video presentation so let's make the requested change optional. 


N.B:  We run quality checks for CR that did not follow the guidelines explicitly, and there is a penalty for such cases.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away :)

Keep rocking 🚀!

Whoistolu reviewed (approved):
### Hi @VitorGuedesMadeira 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VitorGuedesMadeira committed:
Add change

VitorGuedesMadeira committed:
Update README.md

VitorGuedesMadeira committed:
Update README.md

VitorGuedesMadeira committed:
Update README.md

VitorGuedesMadeira committed:
Update README.md
